SimG4Core/Notification/src/GenParticleInfoExtractor.cc

```diff
@@ -3,16 +3,19 @@
 
 const GenParticleInfo &GenParticleInfoExtractor::operator()(const G4PrimaryParticle *p) const {
   G4VUserPrimaryParticleInformation *up = p->GetUserInformation();
-  if (up == nullptr)
-    G4Exception("SimG4Core/Notification",
-                "mc001",
-                FatalException,
-                "GenParticleInfoExtractor: G4PrimaryParticle has no user information");
   GenParticleInfo *gpi = dynamic_cast<GenParticleInfo *>(up);
-  if (gpi == nullptr)
+  if (up == nullptr) {
     G4Exception("SimG4Core/Notification",
                 "mc001",
                 FatalException,
-                "GenParticleInfoExtractor: user information in G4PrimaryParticle is not of GenParticleInfo type");
+                "GenParticleInfoExtractor: G4PrimaryParticle has no user information");
+  } else {
+    GenParticleInfo *gpi = dynamic_cast<GenParticleInfo *>(up);
+    if (gpi == nullptr)
+      G4Exception("SimG4Core/Notification",
+                  "mc001",
+                  FatalException,
+                  "GenParticleInfoExtractor: user information in G4PrimaryParticle is not of GenParticleInfo type");
+  }
   return *gpi;
```

Comment:
Uhm... Now `gpi` is a local variable included in the previous scope.
I wonder what can be returned here

Reply:
There is still `gpi` on this scope on line 6, and line 13 is a duplicate of that, so technically the code does what is intended, but it is confusing. I personally find the earlier version more clear than the changed version.

Reply:
Is fixed now